test/browser/suspense.test.js

```diff
@@ -0,0 +1,235 @@
+	});
+
+	it('should suspend when using lazy', () => {
+		render(<Suspense fallback={<div>Suspended...</div>}>
+			<Lazy />
+		</Suspense>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`<div>Suspended...</div>`
+		);
+
+		return tick(() => {
+			rerender();
+			expect(scratch.innerHTML).to.eql(
+				`<div>Hello Lazy</div>`
+			);
+		});
+	});
+
+	it('should suspend when a promise is throw', () => {
+		render(<Suspense fallback={<div>Suspended...</div>}>
+			<CustomSuspense />
+		</Suspense>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`<div>Suspended...</div>`
+		);
+
+		return tick(() => {
+			rerender();
+			expect(scratch.innerHTML).to.eql(
+				`<div>Hello CustomSuspense...</div>`
+			);
+		});
+	});
+
+	it('should suspend with custom error boundary', () => {
+		render(<Suspense fallback={<div>Suspended...</div>}>
+			<Catcher>
+				<CustomSuspense />
+			</Catcher>
+		</Suspense>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`<div>Suspended...</div>`
+		);
+
+		return tick(() => {
+			rerender();
+			expect(scratch.innerHTML).to.eql(
+				`<div>Hello CustomSuspense...</div>`
+			);
+		});
+	});
+
+	it('should support throwing suspense', () => {
+		render(<Suspense fallback={<div>Suspended...</div>}>
+			<Catcher>
+				<CustomSuspense />
+			</Catcher>
+		</Suspense>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`<div>Suspended...</div>`
+		);
+
+		return tick(() => {
+			rerender();
+			expect(scratch.innerHTML).to.eql(
+				`<div>Hello ThrowingSuspense...</div>`
+			);
+		});
+	});
+
+	it('should only suspend the most inner Suspend', () => {
+		render(<Suspense fallback={<div>Suspended... 1</div>}>
+			Not suspended...
+			<Suspense fallback={<div>Suspended... 2</div>}>
+				<Catcher>
+					<CustomSuspense />
+				</Catcher>
+			</Suspense>
+		</Suspense>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`Not suspended...<div>Suspended... 2</div>`
+		);
+
+		return tick(() => {
+			rerender();
+			expect(scratch.innerHTML).to.eql(
+				`Not suspended...<div>Hello CustomSuspend</div>`
+			);
+		});
+	});
+
+	it('should throw when missing Suspense', () => {
+		render(<Catcher>
+			<CustomSuspense />
+		</Catcher>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`<div>Catcher did catch: CustomSuspense suspended while rendering, but no fallback UI was specified.
+
+Add a &lt;Suspense fallback=...&gt; component higher in the tree to provide a loading indicator or placeholder to display.</div>`
+		);
+	});
+
+	it('should throw when lazy\'s loader throws', () => {
+		render(<Suspense fallback={<div>Suspended...</div>}>
+			<Catcher>
+				<ThrowingLazy />
+			</Catcher>
+		</Suspense>, scratch);
+		rerender();
+		expect(scratch.innerHTML).to.eql(
+			`<div>Suspended...</div>`
+		);
+
+		return tick(() => {
+			rerender();
+			expect(scratch.innerHTML).to.eql(
+				`<div>Hello ThrowingSuspense...</div>`
+			);
+		});
+	});
+});
```

Comment:
nit: missing newline at the end of the file.

Reply:
Should I add an eslint rule for this?